packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -146,7 +146,10 @@ BuildApp() {
     RunCommand cp -r -- "${app_framework}" "${derived_dir}"
 
     StreamOutput " ├─Generating dSYM file..."
-    RunCommand xcrun dsymutil -o "${build_dir}/aot/App.dSYM" "${app_framework}/App"
+    # Placing dSYMs in a folder ending with ".noindex" so Xcode cannot find them
+    # via Spotlight and get confused when uploading an App to the App Store.
+    mkdir "${build_dir}/dSYMs.noindex"
+    RunCommand xcrun dsymutil -o "${build_dir}/dSYMs.noindex/App.framework.dSYM" "${app_framework}/App"
```

Comment:
totally nit-picking: The comment here is confusing to me. It's not a complete sentence. s/Placing/Place/ would fix the grammar, but I think it might be best to phrase the thing backwards: "Xcode uses Spotlight to find files when uploading them to the App store, and gets confused if it finds dSYM files when doing so. To avoid this, we place the dSYM files in a folder ending with ".noindex", which disables spotlight. See: \<link to radar>"

Reply:
Maybe even more specifically:
Xcode uses Spotlight metadata to locate (by GUID) the dSYM associated with each framework embedded in the app to be uploaded to the App Store. To hide the dSYM from spotlight, we place it in a directory ending in '.noindex', which disables spotlight.

Despite the fact this appears to be very intentional on Apple's part, agreed on the radar. Search on [openradar](https://openradar.appspot.com/page/1) first, then file one and copy-paste into openradar if you can't find one.